executor/index_lookup_merge_join.go

```diff
@@ -394,20 +400,27 @@ func (imw *innerMergeWorker) run(ctx context.Context, wg *sync.WaitGroup, cancel
 }
 
 func (imw *innerMergeWorker) handleTask(ctx context.Context, task *lookUpMergeJoinTask) (err error) {
-	numOuterRows := task.outerResult.NumRows()
-	var outerMatch []bool
+	numOuterChks, numOuterRows := task.outerResult.NumChunks(), task.outerResult.Len()
+	var outerMatch [][]bool
 	if imw.outerMergeCtx.filter != nil {
-		outerMatch = make([]bool, numOuterRows)
-		task.memTracker.Consume(int64(cap(outerMatch)))
-		outerMatch, err = expression.VectorizedFilter(imw.ctx, imw.outerMergeCtx.filter, chunk.NewIterator4Chunk(task.outerResult), outerMatch)
-		if err != nil {
-			return err
+		outerMatch = make([][]bool, numOuterChks)
+		for i := 0; i < numOuterChks; i++ {
+			chk := task.outerResult.GetChunk(i)
+			outerMatch[i] = make([]bool, chk.NumRows())
+			outerMatch[i], err = expression.VectorizedFilter(imw.ctx, imw.outerMergeCtx.filter, chunk.NewIterator4Chunk(chk), outerMatch[i])
+			if err != nil {
+				return err
+			}
 		}
 	}
-	task.outerOrderIdx = make([]int, 0, numOuterRows)
-	for i := 0; i < numOuterRows; i++ {
-		if len(outerMatch) == 0 || outerMatch[i] {
-			task.outerOrderIdx = append(task.outerOrderIdx, i)
+	task.outerOrderIdx = make([]chunk.RowPtr, 0, numOuterRows)
```

Comment:
`numOuterRows` is only used here, so we can use `task.outerOrderIdx = make([]chunk.RowPtr, 0, task.outerResult.Len())` here